SingularityService/src/main/java/com/hubspot/singularity/data/curator/ZkClientsLoadDistributor.java

```diff
@@ -41,7 +40,8 @@ public void close() {
   }
 
   public CuratorFramework getCuratorFramework() {
-    return iterator.next();
+    int ci = curatorIndex.getAndUpdate(i -> (i + 1) % curatorFrameworks.size());
```

Comment:
might've been able to get away without using an AtomicInteger here, but don't think it makes a huge difference either way